src/components/MessageInput/hooks/useAttachments.ts

```diff
@@ -12,15 +12,31 @@ import {
   useTranslationContext,
 } from '../../../context';
 
-import type { SendFileAPIResponse } from 'stream-chat';
+import type { Attachment, SendFileAPIResponse } from 'stream-chat';
 import type { MessageInputReducerAction, MessageInputState } from './useMessageInputState';
 import type { MessageInputProps } from '../MessageInput';
-import type { LocalAttachment } from '../types';
+import { AnyLocalAttachment, AttachmentInternalMetadata, LocalAttachment } from '../types';
 import type { FileLike } from '../../ReactFileUtilities';
 import type { CustomTrigger, DefaultStreamChatGenerics } from '../../../types/types';
+import { isUploadedImage } from '../../Attachment';
 
 const apiMaxNumberOfFiles = 10;
 
+const ensureIsLocalAttachment = <
+  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics
+>({
+  localMetadata = {},
+  ...attachment
+}:
+  | Attachment<StreamChatGenerics>
+  | LocalAttachment<StreamChatGenerics>): LocalAttachment<StreamChatGenerics> => ({
```

Comment:
Let's check for `localMetadata.id` before re-creating the object - if it already has that established, it should bail.

Reply:
adjusted